### PR TITLE
Work-around for axa-choice checked attribute controllable by React

### PR DESCRIPTION
**Pull request by pavestru**
Workaround to #439 for axa-choice - does not fix the issue

Changes proposed in this pull request:

 - adds `controlled` attribute to axa-choice
 - adds `controlled` and `uncontrolled` modifiers to `a-choice__input`
 - in case of controlled is set to true, styles are applied based on axa-choice's checked attribute instead of its wrapped input element's checked attribute
 - even though the same was applied to `disabled` attribute, it is discouraged to use controlled disabled attribute, because it is still buggy (it is not possible to enable input once disabled)

## Type of change

- [x] Bug fix (non-breaking change which is just a workaround and not a real fix)
 
 # How Has This Been Tested?
 
Tested on Chrome 66, macOS
 
 # Checklist:
 
 - [ ] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [ ] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/a-choice/_template.js [hunk 1]
```diff
@@ -7,15 +7,16 @@ export default function ({
   name,
   checked = false,
   disabled = false,
+  controlled = false,
 }, childrenFragment) {
   return html`<label class="a-choice__wrapper">
     <input
       id=${inputId}
-      class="a-choice__input" 
+      class="a-choice__input a-choice__input--${controlled ? 'controlled' : 'uncontrolled'}"
```
Comment (AndyOGo) [inline src/components/a-choice/_template.js]: This is indeed a bug.

Though I would not fix the symptom by adding overruling classes.
Rather I assume it to be broken at the our DOM-diffing level.

**Comment by pavestru** [comment]
Yes, @TheDadi , you're right in that that it's not a good solution. It's just my attempt at a workaround in order to move forward until a solution is found. Also thank you for the link to the React issue.

Actually, one of the first things I tried was
```diff
        name="${name}"
-      checked="${checked}"
+      ${checked ? 'checked="checked"' : ''}
       value="${value}"
```
... but it didn't work. Then I realized that this transformation from `checked="true/false"` to `checked="checked"` or no checked attribute displayed at all was already taken care of by nanohtml as it was working correctly at first render. The consecutive re-renders based on the subsequent changed state were not re-rendered correctly.

So, probably using withReact will be the solution as @AndyOGo points out in #439.

**Comment by AndyOGo** [comment]
Released some bugfixes and published a working example at code-sandbox:
https://codesandbox.io/s/o4940jx1y

**Note:** importing polyfills is a must.

**Comment by AndyOGo** [comment]
Luckily we could finally fix this :)

**Comment by pavestru** [comment]
This is super awesome! However, I have to restore the branch as our build process depends on it. I will delete it as soon as we rewrite all our components using `withReact`, so bear with me a little while, please :-)

**Comment by AndyOGo** [comment]
Sorry for being to quick...

**Pull request closed without merging**